community/modules/compute/schedmd-slurm-gcp-v6-nodeset/main.tf

```
@@ -175,3 +175,24 @@ data "google_compute_reservation" "reservation" {
     # Add a validation that if reservation.project != var.project_id it should be a shared reservation
   }
 }
+
+data "google_compute_machine_types" "machine_types_by_zone" {
+  for_each = local.zones
+  filter   = format("name = \"%s\"", var.machine_type)
+  zone     = each.value
+}
+
+locals {
+  machine_types_by_zone   = data.google_compute_machine_types.machine_types_by_zone
+  zones_with_machine_type = [for k, v in local.machine_types_by_zone : k if length(v.machine_types) > 0]
+}
+
+resource "terraform_data" "machine_type_zone_validation" {
+  input = var.machine_type
+  lifecycle {
+    precondition {
+      condition     = length(local.zones_with_machine_type) > 0
+      error_message = "machine type ${var.machine_type} is not available in any of the zones ${jsonencode(local.zones)}"
```

A good suggestion was made to me to help the user to the right answer. I suggest the following:


 to guide the user to select
```suggestion
      error_message = <<-EOT
        machine type ${var.machine_type} is not available in any of the zones ${jsonencode(local.zones)}". To list zones in which it is available, run:

        gcloud compute machine-types list --filter="name=${var.machine_type}"
        EOT
```